File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/PacGoat/src/Commands/CheckForHotGoal.cpp
#include "CheckForHotGoal.h"

#include "Robot.h"

/*
 * All WPILib classes are in the FRC namespace. Either `using namespace frc` 
 * or frc scope (ex. `frc::RobotDrive`) are required to use WPILib functionality. 
 * For headers, use scope instead of `using namespace frc` to avoid global namespace
 * pollution. For source files, either option is viable.
 * See http://wpilib.screenstepslive.com/s/4485/m/13810/l/680719 for more information
 */
using namespace frc;

CheckForHotGoal::CheckForHotGoal(double time) {
	SetTimeout(time);
}

// Called just before this Command runs the first time
void CheckForHotGoal::Initialize() {}

// Called repeatedly when this Command is scheduled to run
void CheckForHotGoal::Execute() {}

// Make this return true when this Command no longer needs to run execute()
bool CheckForHotGoal::IsFinished() {
	return IsTimedOut() || Robot::shooter->GoalIsHot();
}

// Called once after isFinished returns true
void CheckForHotGoal::End() {}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void CheckForHotGoal::Interrupted() {}

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/PacGoat/src/org/usfirst/frc/team190/pacgoat/commands/CheckForHotGoal.java
package $package.commands;

import edu.wpi.first.wpilibj.command.Command;

import $package.Robot;

/**
 * This command looks for the hot goal and waits until it's detected or timed
 * out. The timeout is because it's better to shoot and get some autonomous
 * points than get none. When called sequentially, this command will block until
 * the hot goal is detected or until it is timed out.
 */
public class CheckForHotGoal extends Command {
	public CheckForHotGoal(double time) {
		setTimeout(time);
	}

	// Make this return true when this Command no longer needs to run execute()
	@Override
	protected boolean isFinished() {
		return isTimedOut() || Robot.shooter.goalIsHot();
	}
}
